src/components/PersonInfo/PersonInfo.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import cn from 'classnames';
+import { Person } from '../../types';
+import { PersonLink } from '../PersonLink';
+
+type Props = {
+  person: Person;
+  people: Person[];
+  personId: string | undefined;
+};
+
+export const PersonInfo: React.FC<Props> = ({
+  person,
+  people,
+  personId,
+}) => {
+  let personMother: React.ReactNode = '-';
+  let personFather: React.ReactNode = '-';
+
+  if (person.motherName) {
+    const mother = people
+      .find(m => m.name === person.motherName);
+
+    personMother = typeof mother === 'object'
+      ? <PersonLink person={mother} />
+      : person.motherName;
```

Comment:
It looks a bit complicated. Would be great to prepare array of people before and add mother/father properties in App component

Reply:
Thanks! I created helper function for data preparation and now the logic inside component looks very clear